src/biogeochem/ch4Mod.F90

```diff
@@ -2952,117 +2969,251 @@ subroutine ch4_aere (bounds, num_methc, filter_methc, num_methp, filter_methp, &
          end do
       end do
 
-      diffus_aere = d_con_g(1,1)*1.e-4_r8  ! for CH4: m^2/s
-      ! This parameter is poorly constrained and should be done on a patch-specific basis...
+      allocate(rootfr_vr(nlevsoi))
```

Comment:
Rather than allocate and deallocate every time you do into here. It likely would be better to make these static arrays as module data (since they are small). Then you don't have to allocate and deallocate each time you go inside. You could also make them automatic arrays so you don't have to actually do the allocate and deallocate steps, but making them module data of static size is probably the best here.

Reply:
great idea, will do